object-manipulation-2/src/invert.js

```diff
@@ -0,0 +1,22 @@
+/* exported invert */
+
+// input: (source) -- an object
+// output: a new object with the values and properties of (source) inverted
+
+// -- if source is an empty object
+//     -- return source
+// -- create a variable of an empty object to store inverted object
+// -- loop through the source object
+//     -- add each property value as a property to the empty object, and each property as a value
+// -- return inverted object
+
+function invert(source) {
+  if (source === {}) {
+    return source;
+  }
```

Comment:
This is likely unnecessary. If the object is empty, the `for...in` loop won't start, which should leave you with the same result.